Makefile

```diff
@@ -58,6 +58,18 @@ test-e2e: export OASIS_INDEXER_E2E = true
 test-e2e:
 	@$(GO) test -race -coverpkg=./... -coverprofile=coverage.txt -covermode=atomic -v ./tests/e2e
 
+dump-state: oasis-indexer
+	sed -E -i '' 's/query_on_cache_miss: false/query_on_cache_miss: true/g' "tests/api/api-dev.yml"
+	./oasis-indexer --config tests/api/api-dev.yml
+
+# Run the api tests locally, assuming the environment is set up with an oasis-node that is 
+# accessible as specified in the config file.
+test-api: oasis-indexer
+	sed -E -i '' 's/query_on_cache_miss: true/query_on_cache_miss: false/g' "tests/api/api-dev.yml"
```

Comment:
Here and in dump-state: The analyzer we'll be running on top of this cache is configured by `tests/e2e_regression/e2e_config.yaml`, so the cache prework should happen based on that too. Edit: Oh, I see you changed the paths. I don't agree with the new name: This is an e2e test (analyzer+api), not just an api test. We do already have a couple e2e tests based on synthetic data in `tests/e2e`, so I put the `run.sh` test into `tests/e2e_regression` for disambiguation: The term "regression testing" is often used somewhat loosely to mean testing against (a snapshot of) real-world data and usage patterns.
So please do move `tests/api/*` back under `tests/e2e_regression`. I'm also open to other names if e2e_regression is confusing/misleading, but not this one :)

Thanks for trying to make names make sense though. It's easy to leave behind a naming mess that made sense when you (the generic you; here, Mitja) were writing it, but does not objectively work very well.

Reply:
Thanks for the thorough explanation! I have moved everything from `tests/api/*` to `tests/e2e_regression/*`. I was hesitant to modify the `e2e_regression` dir in case there was a broader plan for it, but I see now that the file-based indexer tests do indeed belong there.